src/chess/pgn.h

```diff
@@ -123,12 +123,78 @@ class PgnReader {
       if (line.empty()) continue;
       std::istringstream iss(line);
       std::string word;
-      while (!iss.eof()) {
+      std::string tmp_word;
+      while (!iss.eof() || !tmp_word.empty()) {
         word.clear();
-        iss >> word;
-        if (word.size() < 2) continue;
+        if (tmp_word.empty()) {
+          iss >> word;
+        } else {
+          word.swap(tmp_word);
+        }
+        if (word[0] == '(') {
+          if (cur_game_.size() == 0) {
+            CERR << "Variation with no move on line " << line_no_ << "!!";
+            throw Exception("Invalid variation.");
+          }
+          variation_stack_.push_back(cur_game_);
```

Comment:
It's totally optional!

but it would really be easier with a recusive function that parses variation, rather than keeping the stack and other state explicitly.

Instead of the outer loop reading tokens, have two functions that can be called from anywhere:
```
std::string PeekToken();
std::string NextToken(); // Reads next and returns it.
void UndoToken(const std::string&); // Not sure if needed.
```

Or even better, not `std::string` tokens as done by std::iostream, but [tokens from the standard](http://www.saremba.de/chessgml/standards/pgn/pgn-complete.htm#c7)

(also I can see that the parser relies on newlines -- actually standard doesn't tell newlines are significant (except as the final character of a single-line comments)).